cmd/juju/common/errors.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"fmt"
+
+	"github.com/juju/errors"
+)
+
+var permMsg = "You do not have permission to %s."
+var grantMsg = `You may ask an administrator to grant you access with "juju grant".`
+
+func PermissionsError(err error, command string) error {
+	if command == "" {
+		command = "complete this operation"
+	}
+	permMsg := fmt.Sprintf(permMsg, command)
+	return errors.Errorf("%v\n\n%s\n%s\n", err, permMsg, grantMsg)
```

Comment:
You should probably wrap the error to keep the stack.
Also, should you check that err is not nil?

```go
newErr := errors.Errorf("%v\n\n%s\n%s\n", err, permMsg, grantMsg)
return errors.Wrap(err, newErr)
```